common/utils/include/claragenomics/utils/allocator.hpp

```diff
@@ -46,6 +47,7 @@ class CudaMallocAllocator
     template <typename U>
     CudaMallocAllocator(const CudaMallocAllocator<U>& rhs)
     {
+        static_cast<void>(rhs);
```

Comment:
Yes, exactly.
I don't think the comment is much help. It is not an idea that requires explanation, but merely a C++ expression. Since the only use of `static_cast<void>(x)` is to avoid unused variable warnings (to my knowledge), and it can be easily found by searching on the internet, this should be fine without a comment.

Reply:
I would still add a short comment `static_cast<void>(rhs); // has no effect, here to suspend unused variable warning`. It simply reduces the number of WTFs_per_line_of_code for less experienced developers

Reply:
Yes, I agree that it is not the most common C++ expression, but it's really just a C++ language thing, nothing elaborate.
The reason I'm hesitant to add the comment is that I will have to add it to every static_cast<void>() then.
It might be a bit awkward the first time you see it, but once you've understood it, all comments are only noise.
If a curious developer stumbles over it, they will look it up, in my opinion.